### PR TITLE
X3D movie texture implementation

### PR DESCRIPTION
**Pull request by mwitchwilliams**
Implementation of X3D &lt;MovieTexture&gt; with interactivity

2 related Pull Requests of tests https://github.com/gearvrf/GearVRf-Tests/pull/258
and the Demo repo gvr-x3d-demo forthcoming

GearVRf-DCO-1.0-Signed-off-by: Mitch Williams
m1.williams@partner.samsung.com

### CODE REVIEW

**Comment by liaxim** [comment]
Needs rebase

File: GVRf/Extensions/x3d/src/main/java/org/gearvrf/x3d/ShaderSettings.java [hunk 1]
```diff
@@ -147,6 +160,26 @@ public String getMaterialName()
...
+  }
+
+  public void setMovieTextureLoop(boolean movieTextureLoop)
+  {
+    this.movieTextureLoop = movieTextureLoop;
+  }
+
+  public boolean getMovieTextureLoop()
+  {
+    return this.movieTextureLoop;
```
Comment (liaxim) [inline GVRf/Extensions/x3d/src/main/java/org/gearvrf/x3d/ShaderSettings.java]: Missing }

**Comment by liaxim** [comment]
If the app is paused, the video player is not paused. Audio can be heard, the cpu usage stays high, etc. You can either handle it in X3D or add a more general pause/resume support to GVRVideoSceneObject. The latter is better but your choice.

**Comment by gearvrf** [comment]
still has conflicts

**Comment by mwitchwilliams** [comment]
Retested with a variety of &lt;TextureTransform&gt; and &lt;MovieTexture&gt; demo's as a final test after the changes were made to resolve the conflicts.

**Comment by liaxim** [comment]
Looks good. Verified by going through the associated x3d test cases and checking out the demos that use the video scene object.

**Comment by liaxim** [comment]
Planning to merge on Friday.

**Pull request merged**